src/python/WMCore/WMRuntime/Bootstrap.py

```diff
@@ -224,6 +224,16 @@ def createInitialReport(job, logLocation):
     report.data.siteName = getattr(siteCfg, 'siteName', 'Unknown')
     report.data.hostName = socket.gethostname()
     report.data.ceName = getSyncCE()
+
+
+    # TODO: need to check what format it returns and what features need to extract.
+    # currently
+    # $MACHINEFEATURES/hs06: HS06 score of the host
+    # $MACHINEFEATURES/total_cpu: number of configured job slots
+    # $JOBFEATURES/hs06_job: HS06 score available to your job
+    # $JOBFEATURES/allocated_cpu: number of allocated slots (=8 in case of a multicore job
+    report.data.machineFeatures = os.environ.get('MACHINEFEATURES', {})
```

Comment:
here I think we need to scan all environment variables ($MACHINEFEATURES/hs06, $MACHINEFEATURES/total_cpu, etc). and then define our own dict which will provide key-value pairs.